pkg/kubectl/service.go

```diff
@@ -134,10 +147,14 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 		if len(portStringSlice) > 1 {
 			name = fmt.Sprintf("port-%d", i+1)
 		}
+		protocol := params["protocol"]
```

Comment:
@AdoHe see https://github.com/kubernetes/kubernetes/pull/24090#issuecomment-217890486 for what I have been thinking about.